osu.Game.Tests/Visual/RealtimeMultiplayer/TestSceneRealtimeRoomManager.cs

```diff
@@ -0,0 +1,153 @@
+                    roomManager.CreateRoom(new Room());
+                    roomManager.PartRoom();
+                    roomManager.CreateRoom(new Room());
+                    roomManager.PartRoom();
+                });
+            });
+
+            AddStep("disconnect", () => roomContainer.Client.Disconnect());
+
+            AddAssert("rooms cleared", () => roomManager.Rooms.Count == 0);
+            AddAssert("initial rooms not received", () => !roomManager.InitialRoomsReceived.Value);
+        }
+
+        [Test]
+        public void TestRoomsPolledOnReconnect()
+        {
+            AddStep("create room manager with a few rooms", () =>
+            {
+                createRoomManager().With(d => d.OnLoadComplete += _ =>
+                {
+                    roomManager.CreateRoom(new Room());
+                    roomManager.PartRoom();
+                    roomManager.CreateRoom(new Room());
+                    roomManager.PartRoom();
+                });
+            });
+
+            AddStep("disconnect", () => roomContainer.Client.Disconnect());
+            AddStep("connect", () => roomContainer.Client.Connect());
+
+            AddAssert("manager polled for rooms", () => roomManager.Rooms.Count == 2);
+            AddAssert("initial rooms received", () => roomManager.InitialRoomsReceived.Value);
+        }
+
+        [Test]
+        public void TestRoomsNotPolledWhenJoined()
+        {
+            AddStep("create room manager with a room", () =>
+            {
+                createRoomManager().With(d => d.OnLoadComplete += _ =>
+                {
+                    roomManager.CreateRoom(new Room());
+                    roomManager.ClearRooms();
+                });
+            });
+
+            AddAssert("manager not polled for rooms", () => roomManager.Rooms.Count == 0);
+            AddAssert("initial rooms not received", () => !roomManager.InitialRoomsReceived.Value);
+        }
+
+        [Test]
+        public void TestMultiplayerRoomJoinedWhenCreated()
+        {
+            AddStep("create room manager with a room", () =>
+            {
+                createRoomManager().With(d => d.OnLoadComplete += _ =>
+                {
+                    roomManager.CreateRoom(new Room());
+                });
+            });
+
+            AddAssert("multiplayer room joined", () => roomContainer.Client.Room != null);
+        }
+
+        [Test]
+        public void TestMultiplayerRoomPartedWhenAPIRoomParted()
+        {
+            AddStep("create room manager with a room", () =>
+            {
+                createRoomManager().With(d => d.OnLoadComplete += _ =>
+                {
+                    roomManager.CreateRoom(new Room());
+                    roomManager.PartRoom();
+                });
+            });
+
+            AddAssert("multiplayer room parted", () => roomContainer.Client.Room == null);
+        }
+
+        [Test]
+        public void TestMultiplayerRoomPartedWhenAPIRoomJoined()
```

Comment:
Not sure why this would be tested this way - being in an API room but not in a multi room seems like an error?

Reply:
This was meant to read "MultiplayerRoomJoinedWhenAPIRoomJoined".